pkg/virt-operator/resource/generate/components/prometheus.go

```diff
@@ -506,6 +506,23 @@ func NewPrometheusRuleSpec(ns string, workloadUpdatesEnabled bool) *v1.Prometheu
 							severityAlertLabelKey: "warning",
 						},
 					},
+					{
+						Record: "kubevirt_vm_container_free_memory_bytes_based_on_requests",
+						Expr:   intstr.FromString("sum by(pod, container) ( kube_pod_container_resource_requests{pod=~'virt-launcher-.*', container='compute', resource='memory'}- on(pod,container) container_memory_working_set_bytes{pod=~'virt-launcher-.*', container='compute'})"),
+					},
+					{
+						Alert: "KubevirtVmHighMemoryUsageBasedOnRequests",
+						Expr:  intstr.FromString("kubevirt_vm_container_free_memory_bytes_based_on_requests < 20971520"),
```

Comment:
Sorry for the delay @sradco 

overall looks good, but if you agree I would say you can extract that `20971520` number to a variable with a meaningful name, it's easy to understand that it represents 20MB in bytes, but I think it would make it more readable

I know its unrelated to Golang but the idea behind this comment is addressed for example in [eslint rules](https://eslint.org/docs/rules/no-magic-numbers#:~:text=Rule%20Details,to%20make%20their%20meaning%20explicit.)

Reply:
After testing this it will complicate the way the expression will look like and it would be clearer to keep as is.